### PR TITLE
Various bug fixes for the DateRange Component

### PR DESCRIPTION
**Pull request by grabkowski**
Fixes in this PR:
FE-2243 DateRange - onChange does not fire correctly
FE-2250 Validations no longer work on date range
FE-2235 Date Range does not work for uncontrolled
FE-2244 DateRange - onBlur does not fire correctly

### CODE REVIEW

File: src/__experimental__/components/date-range/date-range.component.js [hunk 1]
```diff
@@ -32,11 +41,21 @@ class DateRange extends React.Component {
     /** Props for the child start Date component */
     startDateProps: PropTypes.shape({ ...DateInput.propTypes, value: PropTypes.string }),
     /** Props for the child end Date component */
-    endDateProps: PropTypes.shape({ ...DateInput.propTypes, value: PropTypes.string })
+    endDateProps: PropTypes.shape({ ...DateInput.propTypes, value: PropTypes.string }),
+    name: PropTypes.string,
```
Comment (ianforster-sage) [inline src/__experimental__/components/date-range/date-range.component.js]: Please can you add some formatted description comments.

File: src/__experimental__/components/date-range/date-range.component.js [hunk 2]
```diff
@@ -6,18 +6,27 @@ import DateInput from '../date';
 import DateRangeValidator from '../../../utils/validations/date-range';
 import tagComponent from '../../../utils/helpers/tags';
 import StyledDateRange from './date-range.style';
+import DateHelper from '../../../utils/helpers/date';
 
 class DateRange extends React.Component {
+  today = DateHelper.todayFormatted('YYYY-MM-DD');
+
+  isControlled = this.props.value !== undefined;
+
```
[the remark this answers is not in the document]
Comment (edleeks87) [reply src/__experimental__/components/date-range/date-range.component.js]: Yeah agree, I'm going to move the `propTypes` to the bottom of the file as it is how we've done the majority of components and it separates it out better imo

File: src/__experimental__/components/date-range/date-range.component.js [hunk 3]
```diff
@@ -32,11 +41,21 @@ class DateRange extends React.Component {
...
+      formattedValue: DateHelper.formatDateToCurrentLocale(this.startDate),
+      rawValue: DateHelper.formatValue(this.startDate || this.today)
+    },
+    endDateValue: {
+      formattedValue: DateHelper.formatDateToCurrentLocale(this.endDate),
+      rawValue: DateHelper.formatValue(this.endDate || this.today)
+    }
   }
 
   constructor(props) {
```
[the remark this answers is not in the document]
Comment (edleeks87) [reply src/__experimental__/components/date-range/date-range.component.js]: Yeah agreed I've removed it

File: src/utils/helpers/date/date.js [hunk 4]
```diff
@@ -176,7 +176,16 @@ const DateHelper = {
...
   _dateFormats: () => {
     return I18n.t('date.formats.inputs', { defaultValue: DateHelper._defaultDateFormats() });
+  },
+
+  formatDateToCurrentLocale(value) {
+    const defaultDateFormat = 'DD/MM/YYYY';
+
+    const visibleFormat = I18n.t('date.formats.javascript', { defaultValue: defaultDateFormat }).toUpperCase();
+
+    return DateHelper.formatValue(value || DateHelper.todayFormatted(), visibleFormat);
```
[the remark this answers is not in the document]
Comment (edleeks87) [reply src/utils/helpers/date/date.js]: Yeah agree, in both `date` and `date-range` we default to today's date anyway before we pass it to the locale formatter anyway so it's unnecessary here

File: src/__experimental__/components/date-range/date-range.component.js (hunk 1 above)
Comment (edleeks87) [reply src/__experimental__/components/date-range/date-range.component.js]: I assume you mean for `name` and `id`?

**Pull request merged**